File: kakao-ios-sdk-1.0.64/samples/objc/KakaoSample/KakaoSample/StorageViewController.m
/**
 * Copyright 2015 Kakao Corp.
 *
 * Redistribution and modification in source or binary forms are not permitted without specific prior written permission.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

#import "StorageViewController.h"
#import "UIAlertView+Addition.h"
#import "UIImageView+WebCache.h"
#import <KakaoOpenSDK/KakaoOpenSDK.h>

@interface StorageViewController() <UIImagePickerControllerDelegate, UINavigationControllerDelegate>

@property (weak, nonatomic) IBOutlet UIImageView *thumbnailImageView;
@property (weak, nonatomic) IBOutlet UIImageView *profileImageView;
@property (weak, nonatomic) IBOutlet UIImageView *originalImageView;

@end

@implementation StorageViewController {
    UIPopoverController *_popoverController;
}


- (void)viewDidLoad {
    [super viewDidLoad];
}

- (IBAction)upload:(id)sender {
    UIImagePickerController *picker = [[UIImagePickerController alloc] init];
    picker.sourceType = UIImagePickerControllerSourceTypeSavedPhotosAlbum;
    picker.allowsEditing = NO;
    picker.delegate = self;
    
    if ([UIDevice currentDevice].userInterfaceIdiom == UIUserInterfaceIdiomPad) {
        _popoverController = [[UIPopoverController alloc] initWithContentViewController:picker];
        [_popoverController presentPopoverFromRect:CGRectMake(0, 0, 400, 400) inView:self.view permittedArrowDirections:UIPopoverArrowDirectionAny animated:YES];
    } else {
        [self presentViewController:picker animated:YES completion:nil];
    }
}

- (void)imagePickerController:(UIImagePickerController *)picker didFinishPickingMediaWithInfo:(NSDictionary *)info {
    [self dismissImagePickerViewController];
    
    UIImage *image = info[UIImagePickerControllerEditedImage];
    if(!image) {
        image = info[UIImagePickerControllerOriginalImage];
        
        if (image == nil) {
            return;
        }
    }
    
    __weak StorageViewController *wself = self;
    [KOSessionTask storageImageUploadTaskWithImage:image
                                    secureResource:NO
                                 completionHandler:^(KOStorageImageInfo *imageInfo, NSError *error) {
                                     if (error) {
                                         [UIAlertView showMessage:[NSString stringWithFormat:@"upload image error = %@", error.description]];
                                     } else {
                                         [wself.thumbnailImageView sd_setImageWithURL:[NSURL URLWithString:imageInfo.thumbnailImageURL]];
                                         [wself.profileImageView sd_setImageWithURL:[NSURL URLWithString:imageInfo.profileImageURL]];
                                         [wself.originalImageView sd_setImageWithURL:[NSURL URLWithString:imageInfo.originImageURL]];
                                     }
                                 }];
}

- (void)imagePickerControllerDidCancel:(UIImagePickerController *)picker {
    [self dismissImagePickerViewController];
}

- (void)dismissImagePickerViewController {
    if ([UIDevice currentDevice].userInterfaceIdiom == UIUserInterfaceIdiomPad) {
        [_popoverController dismissPopoverAnimated:YES];
        _popoverController = nil;
    } else {
        [self dismissViewControllerAnimated:YES completion:nil];
    }
}

@end

File: kakao-ios-sdk-1.0.64/samples/swift/KakaoSample/KakaoSample/StorageViewController.swift
/**
* Copyright 2015 Kakao Corp.
*
* Redistribution and modification in source or binary forms are not permitted without specific prior written permission.
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*    http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/

import UIKit

class StorageViewController: UIViewController, UIImagePickerControllerDelegate, UINavigationControllerDelegate {

    @IBOutlet weak var thumbnailImageView: UIImageView!
    @IBOutlet weak var profileImageView: UIImageView!
    @IBOutlet weak var originalImageView: UIImageView!
    var popoverController: UIPopoverController!
    
    @IBAction func upload(_ sender: AnyObject) {
        let picker = UIImagePickerController()
        picker.sourceType = UIImagePickerControllerSourceType.savedPhotosAlbum
        picker.allowsEditing = true
        picker.delegate = self
        
        if UIDevice.current.userInterfaceIdiom == UIUserInterfaceIdiom.pad {
            self.popoverController = UIPopoverController(contentViewController: picker)
            self.popoverController?.present(from: CGRect(x: 0, y: 0, width: 400, height: 400), in: self.view, permittedArrowDirections: UIPopoverArrowDirection.any, animated: true)
        } else {
            self.present(picker, animated: true, completion: nil)
        }
    }
    
    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [String : Any]) {
        dismissImagePickerViewController()
        
        var image: UIImage? = info[UIImagePickerControllerEditedImage] as? UIImage
        if image == nil {
            image = info[UIImagePickerControllerOriginalImage] as? UIImage
            
            if image == nil {
                return;
            }
        }
        
        KOSessionTask.storageImageUploadTask(with: image, secureResource: false) { [weak self] (imageInfo, error) -> Void in
            if error != nil {
                UIAlertView.showMessage("upload image failed. \(error)")
            } else {
                if let thumbnailImageURL = URL(string: (imageInfo?.thumbnailImageURL)!) {
                    self?.thumbnailImageView.setImage(withUrl: thumbnailImageURL)
                }
                
                if let profileImageURL = URL(string: (imageInfo?.profileImageURL)!) {
                    self?.profileImageView.setImage(withUrl: profileImageURL)
                }
                
                if let originImageURL = URL(string: (imageInfo?.originImageURL)!) {
                    self?.originalImageView.setImage(withUrl: originImageURL)
                }
            }
        }
    }
    
    func imagePickerControllerDidCancel(_ picker: UIImagePickerController) {
        dismissImagePickerViewController()
    }
    
    fileprivate func dismissImagePickerViewController() {
        if UIDevice.current.userInterfaceIdiom == UIUserInterfaceIdiom.pad {
            self.popoverController?.dismiss(animated: true)
            self.popoverController = nil
        } else {
            self.dismiss(animated: true, completion: nil)
        }
    }
}
